src/Error.cpp

```diff
@@ -63,41 +63,29 @@ InternalError _internal_error("");
 }  // namespace
 
 ErrorReport::ErrorReport(const char *file, int line, const char *condition_string, int flags) : flags(flags) {
-
+    // Note that we deliberately put the entire message into a single line
+    // (aside from newlines inserted by user code) to make it easy to filter
+    // specific warnings or messages via (e.g.) grep.
     const std::string &source_loc = Introspection::get_source_location();
-
+    const char *what = (flags & Warning) ? "Warning" : "Error";
```

Comment:
How about picking a separator like so: 
```
#if defined(WITH_EXCEPTIONS) || defined(_MSC_VER)
const bool use_newlines = false;
#else 
const bool use_newlines = (custom_error_reporter == nullptr) && isatty(2);
#endif  
```

Reply:
why WITH_EXCEPTIONS here?

Reply:
I just looked at the destructor for the circumstances in which something would be printed to stdout. If exceptions are on it throws instead of printing, so isatty is not meaningful.